packages/core-data/src/actions.js

```diff
@@ -273,6 +275,10 @@ export const deleteEntityRecord = (
 			error,
 		} );
 
+		if ( error && throwOnError ) {
```

Comment:
What if `error` is `undefined` or `null` or `0` or any falsy value? Even though it's an edge case, it should still throw such error 😅. I guess we'll need a flag like `hasError` to determine if it reaches the `catch` clause or not.
(Would be awesome if there's a test case for it too, but it's a nitpick.)